### PR TITLE
develop DLARF1F and implement in ORM2R, #1011

### PR DESCRIPTION
**Pull request by EduardFedorenkov**
Hi, @langou!

I develop the first version of DLARF1F.  Some **netlib** tests for QR are failed. Looks like errors in some corner cases.
Please take a look. I will send some description about idea of the algorithm later.

### CODE REVIEW

**Comment by langou** [comment]
Hi @EduardFedorenkov, That's great. Thanks for the work

I think that @jprhyne already started a similar work, so I think it would be good to get in synch. @jprhyne: can you PR your work so that we compare the two PRs?

**Comment by jprhyne** [comment]
Yes, I just made one. 

I have similar issues, and currently am looking at making a small test suite to more easily identify the errors.

**Comment by EduardFedorenkov** [comment]
Looks like I fix the bug. Thanks to @jprhyne PR #1020. @jprhyne handled the case **LASTV = 1** correctly, unlike me!

**Comment by jprhyne** [comment]
Awesome! I'll look at my implementation later today and hopefully fix the issue as well. I'll look at the difference between what we did and hopefully it'll be more obvious then!

@EduardFedorenkov since you wanted to work together, do you want to go ahead and do d/z and I do c/s separately or did you want to make a joint repo and we combine the PRs?

**Comment by EduardFedorenkov** [comment]
Yes, I will take a look at the difference between my and your implementation. Then I think we need to align our versions. I think it is necessary to have almost the same code in {Z,D} and {C,S}. After this it was easy to finish the issue. If you already have some suggestions about the algorithm, please write it in this PR.

**Comment by jprhyne** [comment]
Hello Eduard,

I was thinking that since we are only adding the assumption that v(1) = 1 [or v(lastv)=1], that the general structure could stay similar. I do like the blocking idea on C, but I'm not sure I follow why we need to break it up in 4 sections.

Instead, basically do something like C = [ C1 C2 ] since that reflects that we are breaking up v as 

v = [1 x ... x ]^T  (potentially trailing zeros)

Maybe I'm misunderstanding something that is explained by the 4 block style.

**Comment by jprhyne** [comment]
Hey @EduardFedorenkov, I've merged differences aside from the ones I asked about above, I was wondering if this was an acceptable base for us to start with and then move on to the other precisions or if there were other comments or another preference.

Also, I do not mean to post many comments, just wanted to separate the thoughts into two separate blocks!

**Comment by EduardFedorenkov** [comment]
Hi, Johnathan!

You did the great job! I think now your implementation is totally correct and better than mine!
I checked your realization on my bunch of tests, and it also passed them.

I have some comments about "clean code", but it is OK for now. Let's proceed with your realization.
I align versions with you soon. You did D-precision, so you can proceed with Z.

If you share my opinion, then I would remove my implementation of D and do {C, S} (like you did DLARF1F and DLARF1L).

File: SRC/clarf1l.f [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+*              matrix assuming v(lastv) = 1, where lastv is the last non-zero
+*
+*  =========== DOCUMENTATION ===========
+*
+* Online html documentation available at
+*            http://www.netlib.org/lapack/explore-html/
+*
+*> \htmlonly
+*> Download CLARF1L + dependencies
+*> <a href="http://www.netlib.org/cgi-bin/netlibfiles.tgz?format=tgz&filename=/lapack/lapack_routine/clarf.f">
```
Comment (langou) [inline SRC/clarf1l.f:11]: Hi @EduardFedorenkov and @jprhyne, Can you change this link to clarf1l.f instead of clarf.f? The link won't work, but, next time, we release the link TGZ, ZIP and TXT will be created and the link will work. So we need to have the correct URLs there. And then ditto for each new files. So 1L x 1F x S x C for @EduardFedorenkov and 1L x 1F x D x Z for @jprhyne. Cheers, @langou

Comment (EduardFedorenkov) [reply SRC/clarf1l.f]: done for {S, C}

**Review by langou — APPROVED** [review]
